addon/-private/system/store.js

```diff
@@ -2101,9 +2107,13 @@ Store = Service.extend({
         throw new EmberError(`No model was found for '${modelName}'`);
       }
 
-      assert(`'${inspect(factory)}' does not appear to be an ember-data model`, (typeof factory._create === 'function') );
+      // interopt with the future
+      let klass = getOwner(this).factoryFor ? factory.class : factory;
+
+      assert(`'${inspect(klass)}' does not appear to be an ember-data model`, klass.isModel);
 
-      factory.modelName = factory.modelName || modelName;
+      // TODO: deprecate this
+      klass.modelName = klass.modelName || modelName;
```

Comment:
We don't currently have a path forward yet. I would prefer to deprecate once this is the case.